### PR TITLE
fix issue #90

### PR DESCRIPTION
**Pull request by eviweb**
Signed-off-by: Eric Villard dev@eviweb.fr

### CODE REVIEW

**Comment by rylnd** [comment]
@eviweb is this something that can be documented with a regression test?

**Comment by eviweb** [comment]
@rylnd, yesterday I've updated the travis file to run tests under french and german locale, as I didn't find any easy way to do it differently.
I'm looking for a better solution.

**Comment by eviweb** [comment]
@rylnd, I think this should do the trick.
I created a dedicated file for regression tests to avoid polluting the main specs.
Please let me know if something should be improved.
best regards

**Comment by rylnd** [comment]
@eviweb ah, I forgot about the added locales in the CI matrix. While I would prefer that to explicit regression tests (truer to the problem), it looks like the added locales themselves don't produce a test failure, as shown in [this build](https://travis-ci.org/rylnd/shpec/jobs/126401594). 

I'm guessing that the lack of failing tests there is due to incomplete locale setup in CI (otherwise we wouldn't have #90 in the first place). If you can figure out how to get the locale stuff to generate a failure in the existing test sweet, that would be preferred. If that's not possible, I'd remove the locale stuff from CI and just use the (pretty much analogous) regression tests.

**Comment by eviweb** [comment]
@rylnd, got it ;)
The global var that effectively changes the environment language is LANGUAGE.
I've rebased a [testing branch](https://github.com/eviweb/shpec/tree/travis-locales) on master to be sure to reproduce the issue #90, here is the [result of the build](https://travis-ci.org/eviweb/shpec/jobs/126519454).
I've integrated the fix in this PR.
Is it good for you ?

**Comment by rylnd** [comment]
@eviweb since the minimal steps to reproduce #90 are to set LANGUAGE on CI, I would say that the new regression specs aren't needed (sorry about that). Otherwise, 👍 

@hlangeveld @LeahCim you guys dug into this issue a lot more than I have; is this an acceptable solution to #90?

**Comment by eviweb** [comment]
@rylnd, don't worry about that. It wasn't a waste of time, I've learned something ;)
The more important thing is to solve this issue.
Best regards

File: bin/shpec [hunk 1]
```diff
@@ -49,10 +49,11 @@ it() {
 }
 
 is_function() {
-  case $(type "$1" 2> /dev/null) in
-    (*function*) return 0;;
-  esac
-  return 1
+  if [ "${SHELL}" = "dash" ]; then
```
Comment (rylnd) [inline bin/shpec]: Why does `dash` need a special case here?

File: .travis.yml [hunk 2]
```diff
@@ -25,6 +27,7 @@ script:
     fi
 cache: apt
 before_install:
+  - sudo apt-get update && sudo apt-get --reinstall install -qq language-pack-en language-pack-de language-pack-fr
```
Comment (rylnd) [inline .travis.yml]: Is installing the language packs necessary for reproducing #90?

**Comment by rylnd** [comment]
@eviweb I apologize for letting this sit for so long. Would you be so kind as to rebase this onto master?

File: .travis.yml (hunk 2 above)
Comment (eviweb) [inline .travis.yml]: @rylnd,
yes.
In fact only the english and one non english language pack are needed, I will remove one.

File: bin/shpec (hunk 1 above)
Comment (eviweb) [inline bin/shpec]: @rylnd,
If I'm not wrong `Dash` does not provide the `typeset` command.

**Comment by eviweb** [comment]
@rylnd,
the PR is rebased onto the master branch.
Language packs are currently only installed on linux as I don't know how to install them under OSX.
Best
Eric

**Comment by rylnd** [comment]
Closing in favor of #113 .

**Pull request closed without merging**